File: QViz.NET/Periscope.QViz/JSON/ConfigurationKey.cs
using System.Collections.Generic;

namespace Periscope.QViz.JSON
{
	/// <summary>
	/// JSON Model for Configuration Keys
	/// </summary>
	public class ConfigurationKey
	{
		/// <summary>
		/// Identifier of the Configuration Key Object
		/// </summary>
		public string configurationKeyId;

		/// <summary>
		/// Key of the Configuration
		/// </summary>
		public string configurationKey;

		/// <summary>
		/// Group of the Configuration
		/// </summary>
		public string configurationGroup;

		/// <summary>
		/// Description of the Configuration
		/// </summary>
		public string description;

		/// <summary>
		/// Is this Configuration Active?
		/// </summary>
		public bool isActive;

		/// <summary>
		/// Is this Configuration for a third-party Tool integration?
		/// </summary>
		public bool isTool;

		/// <summary>
		/// Is this Configuration contain any sensitive information?
		/// </summary>
		public bool isSensitiveData;

		/// <summary>
		/// Regular expression for validating the Configuration
		/// </summary>
		public string validationRegex;

		/// <summary>
		/// List of Project Configuration Objects
		/// </summary>
		public IList<ProjectConfiguration> projectConfigurations;

		/// <summary>
		/// Default Constructor
		/// </summary>
		public ConfigurationKey()
		{
			this.configurationKeyId = "";
			this.configurationKey = "";
			this.configurationGroup = "";
			this.description = "";
			this.isActive = false;
			this.isTool = false;
			this.isSensitiveData = false;
			this.validationRegex = "";
			this.projectConfigurations = new List<ProjectConfiguration>();
		}
	}
}

File: QViz.Java/src/main/java/com/periscope/qviz/json/ConfigurationKey.java
package com.periscope.qviz.json;

import java.util.ArrayList;
import java.util.List;

/**
 * JSON Model for Configuration Keys
 */
public class ConfigurationKey {

	/**
	 * Identifier of the Configuration Key Object
	 */
	public String configurationKeyId;

	/**
	 * Key of the Configuration
	 */
	public String configurationKey;

	/**
	 * Group of the Configuration
	 */
	public String configurationGroup;

	/**
	 * Description of the Configuration
	 */
	public String description;

	/**
	 * Is this Configuration Active?
	 */
	public Boolean isActive;

	/**
	 * Is this Configuration for a third-party Tool integration?
	 */
	public Boolean isTool;

	/**
	 * Is this Configuration holds sensitive data
	 */
	public Boolean isSensitiveData;

	/**
	 * Regular expression for validating the Configuration
	 */
	public String validationRegex;

	/**
	 * List of Project Configuration Objects
	 */
	public List<ProjectConfiguration> projectConfigurations;

	/**
	 * Default Constructor
	 */
	public ConfigurationKey() {
		this.configurationKeyId = "";
		this.configurationKey = "";
		this.configurationGroup = "";
		this.description = "";
		this.isActive = false;
		this.isTool = false;
		this.isSensitiveData = false;
		this.validationRegex = "";
		this.projectConfigurations = new ArrayList<>();
	}

}
